scripts/globals/trust.lua

```diff
@@ -28,14 +38,27 @@ tpz.trust.canCast = function(caster, spell, not_allowed_trust_ids)
     -- You can only summon trusts if you are the party leader or solo
     local leader = caster:getPartyLeader()
     if leader and caster:getID() ~= leader:getID() then
-          return tpz.msg.basic.TRUST_NO_CAST_TRUST
+        caster:messageSystem(tpz.msg.system.TRUST_SOLO_OR_LEADER)
+        return -1
     end
 
-    -- TODO: Block summoning trusts if someone recently joined party
+    -- Block summoning trusts if seeking a party
+    if caster:isSeekingParty() then
+        caster:messageSystem(tpz.msg.system.TRUST_NO_SEEKING_PARTY)
+        return -1
+    end
 
+    -- Block summoning trusts if someone recently joined party (120s)
```

Comment:
I don't remember, but are we already blocking party invites to players if trusts are present? (Either a party leader using trusts sending an invite to a free person, or a player-only party sending an invite to someone with trusts out)

Not a requirement for this PR if we're not; just making sure we don't forget any loose threads~

Reply:
Not really, on the core end everything is pretty much unlimited and the only blocking happens on the lua side. The way trusts are handled in and around a player/mixed party is all sorts of wrong. That'll have to be a whole research and dev leg on its own before release :(